src/pages/LoginPage/hooks/useLogin.ts

```diff
@@ -22,12 +25,21 @@ export default function useLogin() {
         console.log('토큰이 없습니다.');
       }
     } catch (err) {
-      console.error('login err', err);
+      setLoading(true);
+
+      setTimeout(() => {
+        setLoading(false);
+        setModalVisible(true);
+      }, 1000);
```

Comment:
여기 로딩 값은 이렇게 하기보다는 try 위에 setLoading(true); 하고 const token = await login(); 아래와 catch문에서 setLoading(false);하면 될 것 같습니다.

```suggestion
      setLoading(false);
      setModalVisible(true);
```

Reply:
try 위에 넣으면 카카오 로그인하기를 눌렀을 때 로딩 화면이 떠서 안에 넣었습니다! 지금은 성공 실패 시에만 로딩 화면이 뜨도록 했는데 버튼을 눌렀을 때도 뜨도록 할까요?

Reply:
아 이건 제가 좀 잘못 전달드린 것 같네요.. 로딩은 저희 서버에 API 요청 보낼 때 뜨도록 의도했습니다.
만약 저희 서버 API가 완료되는데 3초가 걸린다고 가정했을때 토큰을 받아오는 동안 띄워지도록요.

카카오톡 로그인 버튼을 누르면 로딩 없이 바로 카카오톡으로 이동합니다.

카카오톡 로그인 말고 서버 토큰 API 칠때 로딩 입니다.

토큰 발급 요청 -> 로딩
(성공) -> 홈으로 이동
(실패) -> 모달

Reply:
그리고 setTImeout은 잘 안쓰는게 좋을 것 같습니다. 지금 로직을 보면. 서버에서 토큰을 받아온 뒤에 추가로 2초를 더 기다려서 이동하는데요. 그게 아니라 토큰을 받아오는 동안 로딩을 노출하는 것입니다.

Reply:
넵 그럼 서버 API를 사용할 때 로딩화면을 띄우도록 하겠습니다